src/redux/books/books.js

```diff
@@ -0,0 +1,66 @@
+import { createSlice, current } from '@reduxjs/toolkit';
+
+const baseUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi';
+const apiKey = 'av9kWU1C4HkPZEQJTJP0';
+const appUrl = `${baseUrl}/apps/${apiKey}/books`;
+
+const bookSlice = createSlice(
+  {
+    name: 'books',
+    initialState: [],
+    reducers: {
+      addBook: (state, action) => [...current(state), { ...action.payload }],
+      removeBook: (state, action) => current(state).filter((b) => b.item_id !== action.payload),
+      fetchBooks: (state, action) => action.payload,
+    },
+  },
+);
+
+const fetchBooks = async (dispatch) => {
+  const response = await fetch(appUrl);
+  const books = await response.json();
+  const bookList = Object.entries(books).map(([id, [book]]) => (
```

Comment:
Book items get removed from the page when the user refreshes the page. Please make sure Server requests are made with react hooks and data comes from the API and gets stored in redux.

Reply:
Fixed required changes